packages/fxa-auth-server/test/remote/recovery_phone_tests.js

```diff
@@ -114,30 +121,52 @@ describe(`#integration - recovery phone`, function () {
     if (!isTwilioConfigured) {
       this.skip('Invalid twilio accountSid or authToken. Check env / config!');
     }
-    // The phone must be registered first
+
+    // Add totp to account
+    client.totpAuthenticator = new otplib.authenticator.Authenticator();
+    const totpResult = await client.createTotpToken();
+    client.totpAuthenticator.options = {
+      secret: totpResult.secret,
+      crypto: crypto,
+    };
+    await client.verifyTotpCode(client.totpAuthenticator.generate());
+
+    // Add recovery phone
     await client.recoveryPhoneNumberCreate(phoneNumber);
-    const code1 = await redisUtil.recoveryPhone.getCode(client.uid, 0);
-    await client.recoveryPhoneConfirm(code1);
+    await client.recoveryPhoneConfirmSetup(
+      await redisUtil.recoveryPhone.getCode(client.uid)
+    );
 
-    // Now we can send a code
+    // Log back capture session status. It should be unverified, since we
+    // added totp
+    await client.destroySession();
+    client = await Client.login(config.publicUrl, email, password, {
+      version: 'V2',
+    });
+    const sessionStatus1 = await client.sessionStatus();
+
+    // Try sending a recovery phone code and confirming it. This should
+    // Put the session back into a verified state acting as a fallback
+    // for totp
     const sendResp = await client.recoveryPhoneSendCode();
-    const code2 = await redisUtil.recoveryPhone.getCode(client.uid, 1);
-    // TODO: FXA-10945
-    // const confirmResp = await client.recoveryPhoneConfirm(code2);
+    const confirmResp2 = await client.recoveryPhoneConfirmSignin(
+      await redisUtil.recoveryPhone.getCode(client.uid)
+    );
+    const sessionStatus2 = await client.sessionStatus();
 
     assert.equal(sendResp.status, 'success');
-    assert.isDefined(code2);
-    // assert.equal(confirmResp.status, 'success');
-    // TODO: Check session verified.
+    assert.equal(confirmResp2.status, 'success');
+    assert.equal(sessionStatus2.state, 'verified');
+    assert.equal(sessionStatus1.state, 'unverified');
```

Comment:
Nice! I think you could also add a new test case that shows a user can remove totp after getting a verified session.

Reply:
I just added an extra check in this test case. Integration are a bit slower to run, so no need to create a redundant test.